CHANGELOG.next.toml

```diff
@@ -90,3 +90,61 @@ message = "Remove deprecated error kind type aliases."
 references = ["smithy-rs#3189"]
 meta = { "breaking" = true, "tada" = false, "bug" = false, "target" = "client" }
 author = "jdisanti"
+
+[[aws-sdk-rust]]
+message = """
+Unhandled errors have been made opaque to ensure code is written in a future-proof manner. Where previously, you
+might have:
+```rust
+match service_error.err() {
+    GetStorageError::StorageAccessNotAuthorized(_) => { /* ... */ }
+    GetStorageError::Unhandled(unhandled) if unhandled.code() == "SomeUnmodeledErrorCode" {
```

Comment:
code is optional; `== Some(...)`, same for below

Reply:
Good catch!